manager/manager.go

```diff
@@ -37,105 +37,167 @@ import (
-		MaxIdleConnsPerHost:   dc.SenderConfig.NumWorkersPerSender,
-		ResponseHeaderTimeout: dc.SenderConfig.ResponseHeaderTimeout,
-		IdleConnTimeout:       dc.SenderConfig.IdleConnTimeout,
+		MaxIdleConnsPerHost:   dc.NumWorkersPerSender,
+		ResponseHeaderTimeout: dc.ResponseHeaderTimeout,
+		IdleConnTimeout:       dc.IdleConnTimeout,
 	}
 	return transport
 }
 
 // chrysom client listener
 func (m *Manager) Update(items []argus.Item) {
-	recentMap := make(map[string]model.Norn)
-	newNorns := []nornDispatcher{}
-	oldNorns := []dispatch.D{}
+	recentIDMap := make(map[string]model.Norn)
+	recentURLMap := make(map[string]model.Norn)
+
+	newNorns := []filterDispatcher{}
+	oldNorns := []Filterer{}
 
-	transport := NewTransport(m.dispatcherConfig)
+	var dispatcher dispatch.D
+	var filter *dispatch.Filter
+	var url string
 
 	for _, item := range items {
 		id := item.Identifier
 		norn, err := model.ConvertItemToNorn(item)
 		if err != nil {
 			log.WithPrefix(m.logger, level.Key(), level.ErrorValue()).Log(logging.MessageKey(), "failed to convert Item to Norn", "item", item)
 		}
-		if _, ok := m.nornsDispatch[id]; ok {
-			recentMap[id] = norn
+		if (norn.Destination.AWSConfig) == (model.AWSConfig{}) {
+			url = norn.Destination.HttpConfig.URL
 		} else {
-			dispatcher, err := dispatch.NewDispatcher(m.dispatcherConfig, norn, transport)
-			if err == nil {
-				newNorns = append(newNorns, nornDispatcher{norn: norn, dispatcher: dispatcher})
-				recentMap[id] = norn
-			} else {
+			url = norn.Destination.AWSConfig.Sqs.QueueURL
+		}
+
+		if _, ok := m.idFilter[id]; ok {
+			recentIDMap[id] = norn
+		} else {
+			filter, err = dispatch.NewFilter(m.filterConfig, dispatcher, norn, m.sender)
+			if err != nil {
 				m.logger.Log(level.Key(), level.ErrorValue(), logging.MessageKey(), err.Error)
+			} else {
+				recentIDMap[id] = norn
 			}
+
+		}
+
+		if _, ok := m.urlDispatcher[url]; ok {
+			recentURLMap[url] = norn
+		} else {
+			if (norn.Destination.AWSConfig) == (model.AWSConfig{}) {
+				dispatcher, err = dispatch.NewHttpDispatcher(m.dispatcherConfig, norn.Destination.HttpConfig, m.sender, m.logger, m.measures)
+				if err != nil {
+					m.logger.Log(level.Key(), level.ErrorValue(), logging.MessageKey(), err.Error)
+				} else {
+					newNorns = append(newNorns, filterDispatcher{norn: norn, dispatcher: dispatcher, filter: filter})
+					recentURLMap[url] = norn
+				}
+			} else {
+				dispatcher, err = dispatch.NewSqsDispatcher(m.dispatcherConfig, norn.Destination.AWSConfig, m.logger, m.measures)
+				if err != nil {
+					m.logger.Log(level.Key(), level.ErrorValue(), logging.MessageKey(), err.Error)
+				} else {
+					newNorns = append(newNorns, filterDispatcher{norn: norn, dispatcher: dispatcher, filter: filter})
+					recentURLMap[url] = norn
+				}
+			}
+		}
+	}
+
+	for i, filterdis := range m.idFilter {
+		if norn, ok := recentIDMap[i]; !ok {
+			oldNorns = append(oldNorns, filterdis.filter)
+			delete(m.idFilter, i)
```

Comment:
we should only be deleting from this map within a write lock.  Otherwise, we could face a race condition.